### PR TITLE
Move constants defined in "coordinates" sub-package to "sun"

### LINKED ISSUE #3934: Move constants defined in coordinates sub-package to sun
The sunpy.sun sub-package is the central repository for all constants so no constants should be defined outside (or locally). This ensures that different parts of sunpy can use these constants and provides a central place to audit constants and update them as needed.

### PR DESCRIPTION
**Pull request by kakirastern**
### Description

Move some magic numbers from the `coordinates` sub-package to the `sun` sub-package as part of `sunpy.sun.constants`. This is to ensure that different parts of the `sunpy` package can access these constants, while providing a central location for auditing constants and updating them as required.

Fixes #3934

### CODE REVIEW

**Comment by kakirastern** [comment]
Hi @ayshih I have tried my best to make the changes as you have suggested.

**Comment by ayshih** [comment]
Great!  I'll take a close look soon.  But first, some code moved very recently from `sunpy.coordinates.ephemeris` to `sunpy.coordinates.sun`, and it looks like a couple of your changes didn't survive that change.  Please fix the following lines again to use your new constants:

https://github.com/sunpy/sunpy/blob/1bfeed125ac856c038263f1df7edb52b79d6a879/sunpy/coordinates/sun.py#L486

https://github.com/sunpy/sunpy/blob/1bfeed125ac856c038263f1df7edb52b79d6a879/sunpy/coordinates/sun.py#L564

**Comment by kakirastern** [comment]
Thanks @nabobalis!

**Comment by kakirastern** [comment]
> Great! I'll take a close look soon. But first, some code moved very recently from `sunpy.coordinates.ephemeris` to `sunpy.coordinates.sun`, and it looks like a couple of your changes didn't survive that change. Please fix the following lines again to use your new constants:
> 
> https://github.com/sunpy/sunpy/blob/1bfeed125ac856c038263f1df7edb52b79d6a879/sunpy/coordinates/sun.py#L486
> 
> https://github.com/sunpy/sunpy/blob/1bfeed125ac856c038263f1df7edb52b79d6a879/sunpy/coordinates/sun.py#L564

@ayshih Let me double check again and add back those two constants soon to `sunpy.coordinates.sun`.

**Comment by nabobalis** [comment]
I replaced the two numbers in each of those lines with the constants you defined, I am unsure if there is more to be done.

**Comment by kakirastern** [comment]
> I replaced the two numbers in each of those lines with the constants you defined, I am unsure if there is more to be done.

@nabobalis Let me double check just to be sure...

**Comment by kakirastern** [comment]
Should I also fix the malformed table in `sunpy.sun.constants` in the warning of the `ci/circleci: html-docs` test described as follows in the log?:

```
/root/project/.tox/build_docs/lib/python3.7/site-packages/sunpy/sun/constants.py:docstring of sunpy.sun.constants:7: WARNING: Malformed table.
Text in column margin in table line 4.
```

**Comment by nabobalis** [comment]
I think I introduced that with a change, I'll fix it.

**Comment by kakirastern** [comment]
@nabobalis Okay, sure 👍

**Comment by kakirastern** [comment]
Cool, the CI checks are all passing!

File: docs/guide/tour.rst [hunk 1]
```diff
@@ -169,7 +169,10 @@ keys.::
                'metallicity', 'sunspot cycle', 'average intensity',
                'effective temperature', 'mass conversion rate', 'center density',
                'center temperature', 'absolute magnitude', 'mean energy production',
-               'ellipticity', 'GM'])
+               'ellipticity', 'GM', 'true longitude of meridian', 'sidereal rotation rate',
+               'first Carrington rotation', 'first Carrington rotation (JD TT)',
+               'mean synodic period', 'right ascension (RA) of the north pole at epoch J2000.0',
+               'declination of the north pole at epoch J2000.0'])
```
Comment (ayshih) [inline docs/guide/tour.rst:175]:
```suggestion
               'ellipticity', 'GM', 'W_0', 'sidereal rotation rate',
               'first Carrington rotation',
               'mean synodic period', 'alpha_0',
               'delta_0'])
```

File: changelog/4013.feature.rst [hunk 2]
```diff
@@ -0,0 +1,9 @@
+Added extra constants to `sunpy.sun.constants`:
+
+- True longitude of meridian : `sunpy.sun.constants.meridian_lon`
+- Sidereal rotation rate : `sunpy.sun.constants.sidereal_rotation_rate`
+- First Carrington rotation : `sunpy.sun.constants.first_crot_jd`
+- First Carrington rotation (JD TT) : `sunpy.sun.constants.first_crot_jd_tt`
+- Mean synodic period : `sunpy.sun.constants.crot_period`
+- Right Ascension (RA) of the north pole at epoch J2000.0 : `sunpy.sun.constants.sun_north_pole_lon`
+- Declination of the North Pole at epoch J2000.0 : `sunpy.sun.constants.sun_north_pole_lat`
```
Comment (ayshih) [inline changelog/4013.feature.rst:9]:
```suggestion
- Longitude of the prime meridian (epoch J2000.0) : `sunpy.sun.constants.W_0`
- Sidereal rotation rate : `sunpy.sun.constants.sidereal_rotation_rate`
- First Carrington rotation (JD TT) : `sunpy.sun.constants.first_carrington_rotation`
- Mean synodic period : `sunpy.sun.constants.mean_synodic_period`
- Right ascension (RA) of the north pole (epoch J2000.0) : `sunpy.sun.constants.alpha_0`
- Declination of the north pole (epoch J2000.0) : `sunpy.sun.constants.delta_0`
```

File: sunpy/coordinates/sun.py [hunk 3]
```diff
@@ -18,6 +18,7 @@
 from sunpy.time import parse_time
 from sunpy.time.time import _variables_for_parse_time_docstring
 from sunpy.util.decorators import add_common_docstring
+from sunpy.sun.constants import sidereal_rotation_rate, meridian_lon
```
Comment (ayshih) [inline sunpy/coordinates/sun.py:21]:
```suggestion
```

File: sunpy/coordinates/sun.py [hunk 4]
```diff
@@ -94,15 +89,15 @@ def carrington_rotation_time(crot):
     -------
     astropy.time.Time
     """
-    estimate = (_CARRINGTON_ROTATION_PERIOD * (crot - 1)) + _FIRST_CROT_JD
+    estimate = (constants.crot_period.value * (crot - 1)) + constants.first_crot_jd.value
```
Comment (ayshih) [inline sunpy/coordinates/sun.py:92]:
```suggestion
    estimate = (constants.mean_synodic_period.to_value('day') * (crot - 1)) + constants.first_carrington_rotation
```

File: sunpy/coordinates/sun.py [hunk 5]
```diff
@@ -127,7 +122,7 @@ def carrington_rotation_number(t='now'):
     # Estimate the Carrington rotation number by dividing the time that has elapsed since
     # JD 2398167.4 (late in the day on 1853 Nov 9), see Astronomical Algorithms (Meeus 1998, p.191),
     # by the mean synodic period (27.2753 days)
-    estimate = (time.tt.jd - _FIRST_CROT_JD) / _CARRINGTON_ROTATION_PERIOD + 1
+    estimate = (time.tt.jd - constants.first_crot_jd.value) / constants.crot_period.value + 1
```
Comment (ayshih) [inline sunpy/coordinates/sun.py:125]:
```suggestion
    estimate = (time - constants.first_carrington_rotation) / constants.mean_synodic_period + 1
```

File: sunpy/coordinates/sun.py [hunk 6]
```diff
@@ -483,7 +478,7 @@ def _detilt_lon(coord):
 # The longitude in the de-tilted frame of the Sun's prime meridian.
 # The IAU (Seidelmann et al. 2007 and later) defines the true longitude of the meridian (i.e.,
 # without light travel time to Earth and aberration effects) as 84.176 degrees eastward at J2000.
-_DLON_MERIDIAN = Longitude(_detilt_lon(_NODE) + 84.176*u.deg)
+_DLON_MERIDIAN = Longitude(_detilt_lon(_NODE) + meridian_lon)
```
Comment (ayshih) [inline sunpy/coordinates/sun.py:481]:
```suggestion
_DLON_MERIDIAN = Longitude(_detilt_lon(_NODE) + constants.get('W_0'))
```

File: sunpy/coordinates/sun.py [hunk 7]
```diff
@@ -561,7 +556,7 @@ def L0(time='now',
     antetime = (obstime - distance / speed_of_light) if light_travel_time_correction else obstime
 
     # Calculate the de-tilt longitude of the meridian due to the Sun's sidereal rotation
-    dlon_meridian = Longitude(_DLON_MERIDIAN + (antetime - _J2000) * 14.1844*u.deg/u.day)
+    dlon_meridian = Longitude(_DLON_MERIDIAN + (antetime - _J2000) * sidereal_rotation_rate)
```
Comment (ayshih) [inline sunpy/coordinates/sun.py:559]:
```suggestion
    dlon_meridian = Longitude(_DLON_MERIDIAN + (antetime - _J2000) * constants.sidereal_rotation_rate)
```

File: sunpy/coordinates/transformations.py [hunk 8]
```diff
@@ -499,7 +499,8 @@ def _sun_earth_icrf(time):
 # (See Archinal et al. 2011,
 #   "Report of the IAU Working Group on Cartographic Coordinates and Rotational Elements: 2009")
 # The orientation of the north pole in ICRS/HCRS is assumed to be constant in time
-_SOLAR_NORTH_POLE_HCRS = UnitSphericalRepresentation(lon=286.13*u.deg, lat=63.87*u.deg)
+_SOLAR_NORTH_POLE_HCRS = UnitSphericalRepresentation(lon=constants.sun_north_pole_lon,
+                                                     lat=constants.sun_north_pole_lat)
```
Comment (ayshih) [inline sunpy/coordinates/transformations.py:503]:
```suggestion
_SOLAR_NORTH_POLE_HCRS = UnitSphericalRepresentation(lon=constants.get('alpha_0'),
                                                     lat=constants.get('delta_0'))
```

File: sunpy/sun/_constants.py [hunk 9]
```diff
@@ -144,3 +147,45 @@
...
                                     132.712e6, 'km**3 s**-2', 0,
                                     gsfc_fact, system='si')
+
+# true longitude of the meridian (i.e., without light travel time to Earth
+# and aberration effects) is 84.176 degrees eastward at J2000
+physical_constants['true longitude of meridian'] = Constant('',
+                                                            'true longitude of meridian',
+                                                            84.176, 'deg',
+                                                            0, archinal,
+                                                            system='si')
```
Comment (ayshih) [inline sunpy/sun/_constants.py:157]:
```suggestion
# longitude of the prime meridian (without light travel time to Earth
# and aberration effects) is 84.176 degrees eastward at J2000
physical_constants['W_0'] = Constant('W_0',
                                     'longitude of the prime meridian (epoch J2000.0)',
                                     84.176, 'deg',
                                     0, archinal,
                                     system='si')
```

File: sunpy/sun/_constants.py [hunk 10]
```diff
@@ -144,3 +147,45 @@
...
+
+# conversion factor for calculating the de-tilt longitude of the meridian
+# due to the Sun's sidereal rotation
```
Comment (ayshih) [inline sunpy/sun/_constants.py:160]:
```suggestion
# the definitional (fixed) rotation rate of the Sun relative to the stars (i.e., sidereal rotation rate)
```

File: sunpy/sun/_constants.py [hunk 11]
```diff
@@ -144,3 +147,45 @@
...
+                                                           'day', 0, meeus, system='si')
+first_crot_jd_tt = Time(2398167.4, format='jd').tt
+physical_constants['first Carrington rotation (JD TT)'] = Constant('',
+                                                                   'first Carrington '
+                                                                   'rotation (JD TT)',
+                                                                   first_crot_jd_tt.value,
+                                                                   'day', 0,
+                                                                   meeus, system='si')
+
+# length of a Carrington rotation in days
```
Comment (ayshih) [inline sunpy/sun/_constants.py:177]:
```suggestion
# length of the mean Carrington rotation as seen from Earth
```

File: sunpy/sun/_constants.py [hunk 12]
```diff
@@ -144,3 +147,45 @@
...
+
+# Sun's north pole is oriented RA=286.13 deg, Dec=63.87 deg in ICRS and HCRS
+physical_constants['right ascension (RA) of the north pole at epoch J2000.0'] \
+    = Constant('alpha_0',
+               'right ascension (RA) of the north pole at epoch J2000.0',
+               286.13, 'deg', 0, archinal, system='si')
+physical_constants['declination of the north pole at epoch J2000.0'] \
+    = Constant('delta_0',
+               'declination of the north pole at epoch J2000.0',
+               63.87, 'deg', 0, archinal, system='si')
```
Comment (ayshih) [inline sunpy/sun/_constants.py:191]:
```suggestion
physical_constants['alpha_0'] \
    = Constant('alpha_0',
               'right ascension (RA) of the north pole (epoch J2000.0)',
               286.13, 'deg', 0, archinal, system='si')
physical_constants['delta_0'] \
    = Constant('delta_0',
               'declination of the north pole (epoch J2000.0)',
               63.87, 'deg', 0, archinal, system='si')
```

File: sunpy/sun/constants.py [hunk 13]
```diff
@@ -121,3 +122,10 @@ def print_all():
 sfu = get('solar flux unit')
 # Observable parameters
 average_angular_size = get('average angular size')
+meridian_lon = get('true longitude of meridian')
+sidereal_rotation_rate = get('sidereal rotation rate')
+first_crot_jd = get('first Carrington rotation')
+first_crot_jd_tt = get('first Carrington rotation (JD TT)')
+crot_period = get('mean synodic period')
+sun_north_pole_lon = get('right ascension (RA) of the north pole at epoch J2000.0')
+sun_north_pole_lat = get('declination of the north pole at epoch J2000.0')
```
Comment (ayshih) [inline sunpy/sun/constants.py:131]:
```suggestion
sidereal_rotation_rate = get('sidereal rotation rate')
first_carrington_rotation = Time(get('first Carrington rotation (JD TT)'), format='jd', scale='tt')
mean_synodic_period = get('mean synodic period')
```

File: sunpy/sun/tests/test_constants.py [hunk 14]
```diff
@@ -8,13 +8,13 @@
 
 def test_find_all():
     assert isinstance(con.find(), list)
-    assert len(con.find()) == 28
+    assert len(con.find()) == 35
```
Comment (ayshih) [inline sunpy/sun/tests/test_constants.py:11]:
```suggestion
    assert len(con.find()) == 34
```

File: sunpy/sun/tests/test_constants.py [hunk 15]
```diff
@@ -8,13 +8,13 @@
...
 
 
 def test_print_all():
     table = con.print_all()
     assert isinstance(table, Table)
-    assert len(table) == 28
+    assert len(table) == 35
```
Comment (ayshih) [inline sunpy/sun/tests/test_constants.py:17]:
```suggestion
    assert len(table) == 34
```

File: sunpy/sun/_constants.py [hunk 16]
```diff
@@ -144,3 +147,45 @@
...
+# due to the Sun's sidereal rotation
+physical_constants['sidereal rotation rate'] = Constant('', 'sidereal rotation rate',
+                                                        14.1844, 'deg day**-1', 0,
+                                                        archinal, system='si')
+
+# time in Julian Days (TT) of the start of the first Carrington rotation
+physical_constants['first Carrington rotation'] = Constant('', 'first Carrington rotation',
+                                                           2398167.4,
+                                                           'day', 0, meeus, system='si')
+first_crot_jd_tt = Time(2398167.4, format='jd').tt
```
Comment (ayshih) [inline sunpy/sun/_constants.py:169]:
```suggestion
first_carrington_rotation = Time(2398167.4, format='jd', scale='tt')
```

File: sunpy/sun/_constants.py [hunk 17]
```diff
@@ -144,3 +147,45 @@
...
+physical_constants['first Carrington rotation (JD TT)'] = Constant('',
+                                                                   'first Carrington '
+                                                                   'rotation (JD TT)',
+                                                                   first_crot_jd_tt.value,
```
Comment (ayshih) [inline sunpy/sun/_constants.py:173]:
```suggestion
                                                                   first_carrington_rotation.tt.jd,
```

File: sunpy/sun/_constants.py [hunk 18]
```diff
@@ -144,3 +147,45 @@
...
+                                                                   'day', 0,
+                                                                   meeus, system='si')
+
+# length of a Carrington rotation in days
+physical_constants['mean synodic period'] = Constant('',
+                                                     'mean synodic period',
+                                                     27.2753, 'day', 0,
+                                                     allen, system='si')
```
Comment (ayshih) [inline sunpy/sun/_constants.py:181]:
```suggestion
                                                     aa, system='si')
```

File: sunpy/sun/_constants.py [hunk 19]
```diff
@@ -12,8 +13,10 @@
 # references
 gsfc_fact = "https://nssdc.gsfc.nasa.gov/planetary/factsheet/sunfact.html"
 allen = "Allen's Astrophysical Quantities 4th Ed."
```
Comment (ayshih) [inline sunpy/sun/_constants.py:15]:
```suggestion
aa = "Astronomical Almanac (2013 edition)"
allen = "Allen's Astrophysical Quantities 4th Ed."
```

File: sunpy/sun/constants.py [hunk 20]
```diff
@@ -9,7 +9,8 @@
     'get', 'find', 'print_all', 'spectral_classification', 'au', 'mass', 'equatorial_radius',
     'volume', 'surface_area', 'average_density', 'equatorial_surface_gravity',
     'effective_temperature', 'luminosity', 'mass_conversion_rate', 'escape_velocity', 'sfu',
-    'average_angular_size'
+    'average_angular_size', 'meridian_lon', 'sidereal_rotation_rate', 'first_crot_jd',
+    'first_crot_jd_tt', 'crot_period', 'sun_north_pole_lon', 'sun_north_pole_lat'
```
Comment (ayshih) [inline sunpy/sun/constants.py:13]:
```suggestion
    'average_angular_size', 'sidereal_rotation_rate', 'first_carrington_rotation',
    'mean_synodic_period'
```

File: sunpy/coordinates/sun.py [hunk 21]
```diff
@@ -94,15 +89,15 @@ def carrington_rotation_time(crot):
     -------
     astropy.time.Time
     """
-    estimate = (_CARRINGTON_ROTATION_PERIOD * (crot - 1)) + _FIRST_CROT_JD
+    estimate = (constants.crot_period.value * (crot - 1)) + constants.first_crot_jd.value
 
     # The above estimate is inaccurate (see comments below in carrington_rotation_number),
     # so put the estimate into carrington_rotation_number to determine a correction amount
     def refine(estimate):
         crot_estimate = carrington_rotation_number(t=Time(estimate, scale='tt', format='jd'))
```
Comment (ayshih) [inline sunpy/coordinates/sun.py:97]:
```suggestion
        crot_estimate = carrington_rotation_number(estimate)
```

File: sunpy/coordinates/sun.py [hunk 22]
```diff
@@ -94,15 +89,15 @@ def carrington_rotation_time(crot):
...
         dcrot = crot - crot_estimate
         # Correct the estimate using a linear fraction of the Carrington rotation period
-        return estimate + (dcrot * _CARRINGTON_ROTATION_PERIOD)
+        return estimate + (dcrot * constants.crot_period.value)
```
Comment (ayshih) [inline sunpy/coordinates/sun.py:100]:
```suggestion
        return estimate + (dcrot * mean_synodic_period)
```
Also change the last line of this function to simply
```
    return estimate
```

**Comment by nabobalis** [comment]
A missing time import is causing it to fail, I will fix that soon(TM).

File: sunpy/coordinates/tests/test_sun.py [hunk 23]
```diff
@@ -467,7 +469,7 @@ def test_carrington_rotation_starttime(crot, julian_days):
 def test_carrington_rotation_roundtrip():
     t = Time('2010-1-1')
     crot = sun.carrington_rotation_number(t)
-    t_roundtrip = sun.carrington_rotation_time(crot)
+    t_roundtrip = sun.carrington_rotation_time(crot.value)
```
Comment (ayshih) [inline sunpy/coordinates/tests/test_sun.py:472]:
```suggestion
    t_roundtrip = sun.carrington_rotation_time(crot)
```
Oops!  I hadn't intended to change the return type of `sunpy.coordinates.sun.carrington_rotation_number()`.  Can you revert this change here and change the return statement in `sunpy.coordinates.sun.carrington_rotation_number()` to `return actual.to_value(u.one)`?

Comment (nabobalis) [reply sunpy/coordinates/tests/test_sun.py]: Fixed.

File: changelog/4013.feature.rst [hunk 24]
```diff
@@ -0,0 +1,8 @@
+Added extra constants to `sunpy.sun.constants`:
+
+- Longitude of the prime meridian (epoch J2000.0) : `sunpy.sun.constants.W_0`
```
Comment (ayshih) [inline changelog/4013.feature.rst:3]:
```suggestion
- Longitude of the prime meridian (epoch J2000.0) : ``sunpy.sun.constants.get('W_0')``
```

File: changelog/4013.feature.rst [hunk 25]
```diff
@@ -0,0 +1,8 @@
...
+- Sidereal rotation rate : `sunpy.sun.constants.sidereal_rotation_rate`
+- First Carrington rotation (JD TT) : `sunpy.sun.constants.first_carrington_rotation`
+- Mean synodic period : `sunpy.sun.constants.mean_synodic_period`
+- Right ascension (RA) of the north pole (epoch J2000.0) : `sunpy.sun.constants.alpha_0`
+- Declination of the north pole (epoch J2000.0) : `sunpy.sun.constants.delta_0`
```
Comment (ayshih) [inline changelog/4013.feature.rst:8]:
```suggestion
- Right ascension (RA) of the north pole (epoch J2000.0) : ``sunpy.sun.constants.get('alpha_0')``
- Declination of the north pole (epoch J2000.0) : ``sunpy.sun.constants.get('delta_0')``
```

File: sunpy/sun/_constants.py [hunk 26]
```diff
@@ -144,3 +148,41 @@
...
+                                                        14.1844, 'deg day**-1', 0,
+                                                        archinal, system='si')
+
+# time in Julian Days (TT) of the start of the first Carrington rotation
+first_carrington_rotation = Time(2398167.4, format='jd', scale='tt')
+physical_constants['first Carrington rotation (JD TT)'] = Constant('',
+                                                                   'first Carrington '
+                                                                   'rotation (JD TT)',
+                                                                   first_carrington_rotation.tt.jd,
+                                                                   'day', 0,
```
Comment (ayshih) [inline sunpy/sun/_constants.py:171]:
```suggestion
                                                                   'day', 0.1,
```

File: sunpy/sun/_constants.py [hunk 27]
```diff
@@ -144,3 +148,41 @@
...
+                                                                   meeus, system='si')
+
+# length of the mean Carrington rotation as seen from Earth
+physical_constants['mean synodic period'] = Constant('',
+                                                     'mean synodic period',
+                                                     27.2753, 'day', 0,
+                                                     aa, system='si')
```
Comment (ayshih) [inline sunpy/sun/_constants.py:178]:
```suggestion
# the rotation rate of the Sun appears to be slower by the rate at which the Earth orbits the Sun
period = 1 / (physical_constants['sidereal_rotation_rate'] / 360*u.deg - 1 / u.yr)
physical_constants['mean synodic period'] = Constant('',
                                                     'mean synodic period',
                                                     period.to_value('day'), 'day', 0,
                                                     archinal, system='si')
```

File: sunpy/sun/_constants.py [hunk 28]
```diff
@@ -4,16 +4,20 @@
...
 from astropy.constants import Constant
+from astropy.time import Time
 
 __all__ = ['physical_constants']
 
 physical_constants = {}
 
 # references
 gsfc_fact = "https://nssdc.gsfc.nasa.gov/planetary/factsheet/sunfact.html"
+aa = "Astronomical Almanac (2013 edition)"
```
Comment (ayshih) [inline sunpy/sun/_constants.py:15]:
```suggestion
```

File: sunpy/sun/_constants.py (hunk 27 above)
Comment (nabobalis) [reply sunpy/sun/_constants.py]: This did not work.
```
astropy.units.core.UnitConversionError: Can only apply 'subtract' function to quantities with compatible dimensions
```

Comment (ayshih) [reply sunpy/sun/_constants.py]:
```suggestion
# the rotation rate of the Sun appears to be slower by the rate at which the Earth orbits the Sun
period = 1 / (physical_constants['sidereal_rotation_rate'] / (360*u.deg) - 1 / u.yr)
physical_constants['mean synodic period'] = Constant('',
                                                     'mean synodic period',
                                                     period.to_value('day'), 'day', 0,
                                                     archinal, system='si')
```
Sorry, I left off parentheses.

File: sunpy/sun/_constants.py [hunk 29]
```diff
@@ -144,3 +148,43 @@
...
+physical_constants['first Carrington rotation (JD TT)'] = Constant('',
+                                                                   'first Carrington '
+                                                                   'rotation (JD TT)',
+                                                                   first_carrington_rotation.tt.jd,
+                                                                   'day', 0.1,
+                                                                   meeus, system='si')
+
+# length of the mean Carrington rotation as seen from Earth
+# the rotation rate of the Sun appears to be slower by the rate at which the Earth orbits the Sun
+period = 1 / (physical_constants['sidereal rotation rate'] / 360*u.deg - 1 / u.yr)
```
Comment (ayshih) [inline sunpy/sun/_constants.py:176]:
```suggestion
period = 1 / (physical_constants['sidereal rotation rate'] / (360*u.deg) - 1 / u.yr)
```

Comment (nabobalis) [reply sunpy/sun/_constants.py]: Thanks.

**Comment by nabobalis** [comment]
Thanks everyone.
